packages/nx/src/project-graph/plugins/loaded-nx-plugin.ts

```diff
@@ -59,19 +59,6 @@ export class LoadedNxPlugin {
       this.exclude = pluginDefinition.exclude;
     }
 
-    if (plugin.createNodes && !plugin.createNodesV2) {
```

Comment:
We should throw an error here to let plugin authors know that their plugin is incompatible with Nx 21. They have already been notified via tsdoc deprecations.